lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -202,7 +209,33 @@ export default function BasicLtiLaunchApp({ rpcServer }) {
 
     try {
       await authWindow.current.authorize();
-      await Promise.all([fetchContentUrl(), fetchGroups()]);
+      // In the re-try case, block only the contentUrl request. At
```

Comment:
I think this block could be implemented in a simpler way. Supposing the `fetchContentUrl` and `fetchGroups` functions were changed to return `true` on success as suggested above:

```js
const [fetchedContent, fetchedGroups] = Promise.all([fetchContentUrl(), fetchGroups()]);
if (fetchedContent && fetchedGroups) {
  setErrorState(null);
}
```

An alternative approach would be for `fetchContentUrl` and `fetchGroups` to re-throw any error that occurs. Then in this block the code would just be:

```js
try {
  await authWindow.current.authorize();
  await Promise.all([fetchContentUrl(), fetchGroups()]);
  setErrorState(null);
} catch (e) {
 // Ignore error here.
} finally {
  authWindow.current = null;
}
```

If you went for this second route, you'd also have to ignore any exceptions when `fetchContentUrl` and `fetchGroups` are called after the component is initially rendered.

Reply:
@robertknight , just wanted to get some clarification on this block once more.

Just glancing at your suggestions here.

**Suggestion A.**
This solution appears to not block on either request. Is that correct? Was there supposed to be an `await` before Promise.all ? If they don't block, then both will return right away so I'm a bit confused by that.

**Suggestion B.**
This solution appears to block on both requests, but this is different than what I have done.  In my solution, I specifically did not block on the groups request --only on the contentUrl. This means if the groups  request take a long time, and the contentUrl does not, then we arrive at the promise `then` block for the contentUrl's promise before the groups are done. Presumable, the return code for groups (`groupError`) would still be false and we would happily set error to null.

After that point, two outcomes may occur a few moments later on. 1. The groups finishes w/o an error, no UI changes occur. 2. The groups call errors,  sets an error state, and the popup jumps right back on screen. 

The other path this code handles is if the groups call actaul errors before the contentUrl. In that case, we use the variable `groupError` and check that value in the contentUrl's `then` block. In that case we know not to clear the error state.

I'm a bit confused how the examples you proposed handle that. But maybe I'm way overthinking this code block or just not understanding something that could be a whole lot more simple and I apologize for that.

Reply:
> Was there supposed to be an await before Promise.all ?

Yes, indeed there was. It should have been `await Promise.all([...])`.